### ISSUE TITLE
X server freezes when executing TF models mnist example

### ISSUE THREAD

**Issue by csuji** [ROCmSoftwarePlatform/tensorflow#13]
after installing tensorflow 1.3 on rocm 1.7.1 and  running 
python convolutional.py

desktop freezes when reaching step tuning backfilters. 
When logging in via ssh and killing the python process below stack trace is logged to kern.log
Desktop remains unresponsive (shell works) and only hard reset  brings back X server.

Mar 21 22:19:53 beastieboys kernel: [ 7371.536475] INFO: task amdgpu_cs:0:1883 blocked for more than 120 seconds.
Mar 21 22:19:53 beastieboys kernel: [ 7371.536485]       Tainted: G           OE   4.13.0-36-generic #40~16.04.1-Ubuntu
Mar 21 22:19:53 beastieboys kernel: [ 7371.536488] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Mar 21 22:19:53 beastieboys kernel: [ 7371.536493] amdgpu_cs:0     D    0  1883   1653 0x00000000
Mar 21 22:19:53 beastieboys kernel: [ 7371.536499] Call Trace:
Mar 21 22:19:53 beastieboys kernel: [ 7371.536512]  __schedule+0x3d6/0x8b0
Mar 21 22:19:53 beastieboys kernel: [ 7371.536518]  schedule+0x36/0x80
Mar 21 22:19:53 beastieboys kernel: [ 7371.536647]  amd_sched_entity_push_job+0xc9/0x110 [amdgpu]
Mar 21 22:19:53 beastieboys kernel: [ 7371.536655]  ? wait_woken+0x80/0x80
Mar 21 22:19:53 beastieboys kernel: [ 7371.536774]  amdgpu_job_submit+0x6c/0x80 [amdgpu]
Mar 21 22:19:53 beastieboys kernel: [ 7371.536865]  amdgpu_vm_bo_update_mapping+0x324/0x3f0 [amdgpu]
Mar 21 22:19:53 beastieboys kernel: [ 7371.536954]  ? amdgpu_vm_free_mapping.isra.20+0x30/0x30 [amdgpu]
Mar 21 22:19:53 beastieboys kernel: [ 7371.537039]  amdgpu_vm_clear_freed+0xae/0x180 [amdgpu]
Mar 21 22:19:53 beastieboys kernel: [ 7371.537123]  amdgpu_gem_va_ioctl+0x4d0/0x520 [amdgpu]
Mar 21 22:19:53 beastieboys kernel: [ 7371.537207]  ? amdgpu_gem_metadata_ioctl+0x1c0/0x1c0 [amdgpu]
Mar 21 22:19:53 beastieboys kernel: [ 7371.537235]  drm_ioctl_kernel+0x6b/0xb0 [drm]
Mar 21 22:19:53 beastieboys kernel: [ 7371.537257]  ? drm_ioctl_kernel+0x6b/0xb0 [drm]
Mar 21 22:19:53 beastieboys kernel: [ 7371.537278]  drm_ioctl+0x3e4/0x450 [drm]
Mar 21 22:19:53 beastieboys kernel: [ 7371.537359]  ? amdgpu_gem_metadata_ioctl+0x1c0/0x1c0 [amdgpu]
Mar 21 22:19:53 beastieboys kernel: [ 7371.537367]  ? kmem_cache_free+0x1b2/0x1c0
Mar 21 22:19:53 beastieboys kernel: [ 7371.537373]  ? kmem_cache_free+0x1b2/0x1c0
Mar 21 22:19:53 beastieboys kernel: [ 7371.537449]  amdgpu_drm_ioctl+0x4c/0x80 [amdgpu]
Mar 21 22:19:53 beastieboys kernel: [ 7371.537457]  do_vfs_ioctl+0xa4/0x600
Mar 21 22:19:53 beastieboys kernel: [ 7371.537463]  ? ____fput+0xe/0x10
Mar 21 22:19:53 beastieboys kernel: [ 7371.537467]  ? task_work_run+0x88/0xb0
Mar 21 22:19:53 beastieboys kernel: [ 7371.537474]  SyS_ioctl+0x79/0x90
Mar 21 22:19:53 beastieboys kernel: [ 7371.537481]  entry_SYSCALL_64_fastpath+0x24/0xab
Mar 21 22:19:53 beastieboys kernel: [ 7371.537485] RIP: 0033:0x7f12d6e15f47
Mar 21 22:19:53 beastieboys kernel: [ 7371.537488] RSP: 002b:00007f12cca969f8 EFLAGS: 00000246 ORIG_RAX: 0000000000000010
Mar 21 22:19:53 beastieboys kernel: [ 7371.537493] RAX: ffffffffffffffda RBX: 00007f12ac009470 RCX: 00007f12d6e15f47
Mar 21 22:19:53 beastieboys kernel: [ 7371.537496] RDX: 00007f12cca96a40 RSI: 00000000c0286448 RDI: 000000000000000d
Mar 21 22:19:53 beastieboys kernel: [ 7371.537498] RBP: 00007f12cca96a90 R08: 0000000115eb0000 R09: 000000000000000e
Mar 21 22:19:53 beastieboys kernel: [ 7371.537501] R10: 0000000000000002 R11: 0000000000000246 R12: 0000000040086409
Mar 21 22:19:53 beastieboys kernel: [ 7371.537504] R13: 000000000000000d R14: 00005624b2996798 R15: 00007f12ac0090c0

### System information
- Ubuntu 16.04.4 LTS
- Linux beastieboys 4.13.0-36-generic #40
- X.Org X Server 1.18.4
- AMD rocm 1.7.1
- Vega FE

**Comment by whchung** [comment]
@csuji in ROCm 1.7.1 a kernel parameter `noretry` has been set to 1 to improve overall system performance. However it has been proven to bring instability to graphics driver shipped with Ubuntu. This is an ongoing issue and we are looking into it.

Before that, please try apply this change by changing `noretry` bit to 0.

```
echo 0 | sudo tee /sys/module/amdkfd/parameters/noretry
```

**Comment by csuji** [comment]
Thanks! Now it worked!

2018-03-21 23:08:57.891523: W tensorflow/stream_executor/rocm/rocm_driver.cc:405] creating context when one is currently active; existing: 0x7f5066a8c7f0
2018-03-21 23:08:57.891609: I tensorflow/core/common_runtime/gpu/gpu_device.cc:907] Found device 0 with properties: 
name: Device 6863
AMDGPU ISA: gfx900
memoryClockRate (GHz) 1.6
pciBusID 0000:03:00.0
Total memory: 15.98GiB
Free memory: 15.73GiB
2018-03-21 23:08:57.891618: I tensorflow/core/common_runtime/gpu/gpu_device.cc:929] DMA: 0 
2018-03-21 23:08:57.891622: I tensorflow/core/common_runtime/gpu/gpu_device.cc:939] 0:   Y 
2018-03-21 23:08:57.891626: I tensorflow/core/common_runtime/gpu/gpu_device.cc:997] Creating TensorFlow device (/gpu:0) -> (device: 0, name: Device 6863, pci bus id: 0000:03:00.0)
2018-03-21 23:08:59.117480: I tensorflow/core/kernels/conv_ops.cc:670] running auto-tune for Convolve
2018-03-21 23:08:59.448617: I tensorflow/core/kernels/conv_ops.cc:670] running auto-tune for Convolve
2018-03-21 23:08:59.513734: I tensorflow/core/kernels/conv_grad_input_ops.cc:858] running auto-tune for Backward-Data
2018-03-21 23:08:59.553221: I tensorflow/core/kernels/conv_grad_filter_ops.cc:778] running auto-tune for Backward-Filter
2018-03-21 23:08:59.588255: I tensorflow/core/kernels/conv_grad_input_ops.cc:858] running auto-tune for Backward-Data
2018-03-21 23:08:59.619737: I tensorflow/core/kernels/conv_grad_filter_ops.cc:778] running auto-tune for Backward-Filter
Extracting data/train-images-idx3-ubyte.gz
Extracting data/train-labels-idx1-ubyte.gz
Extracting data/t10k-images-idx3-ubyte.gz
Extracting data/t10k-labels-idx1-ubyte.gz
Initialized!
Step 0 (epoch 0.00), 17.7 ms
Minibatch loss: 8.334, learning rate: 0.010000
Minibatch error: 85.9%
Validation error: 84.6%
Step 100 (epoch 0.12), 6.3 ms
Minibatch loss: 3.250, learning rate: 0.010000
Minibatch error: 6.2%
......

**Comment by whchung** [comment]
@csuji Glad to hear that, files under `/sys` won't be preserved after reboot. One way to keep it is to change `/etc/modprobe.d/amdkfd.conf` and make it be:

```
options amdkfd noretry=0
```

Once it's done, run `sudo update-initramfs -u`. Reboot and verify `/sys/module/amdkfd/parameters/noretry` stays as 0.

**Comment by csuji** [comment]
How big is the performance penalty? Because with other OpenCL programs I did not have this problem yet.

**Comment by sunway513** [comment]
The performance penalty depends on the networks you use, overall you can expect less than 10% performance drop.

**Comment by whchung** [comment]
Need @dagamayank to provide more detailed information. Basically it affects hand-written GCN assembly kernels with interleaving SGPR loads in `MIOpen`, winograd convolution is one example. For kernels written in high level languages such as OpenCL, HIP, or C++ they won't be affected as the compiler emits SGPRs loads in the beginning.

**Comment by csuji** [comment]
Okay, thanks. Looks negligible.

**Comment by dagamayank** [comment]
VGG/OxfordNet will have the most performance impact, others should be within 10%.
